cpacker/MemGPT
# feat: Migration tool for agents pre DB refactor (pre 0.2.12)

**Please describe the purpose of this pull request.**

- [ ] Add migration tool that automatically converts old agent folders into the new format
- [ ] Make migration UX nice
  - on the first run of the new refactor release, users should see a prompt asking them if they want to migrate old agents
  - for the initial release past the refactor, should also add a "Don't see old agents? Make sure you run the migration command" hint to the agent selection screen

**How to test**

- Try migrating old agent folders, manually inspect the new data to make sure it matches

**Have you tested this PR?**

WIP

**Related issues or PRs**

#708
